services/github/github-directory-file-count.tester.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const t = (module.exports = require('../tester').createServiceTester())
+
+const mockContents = nock => {
+  const contents = [
+    { path: 'a', type: 'dir' },
+    { path: 'b', type: 'dir' },
+    { path: 'c.js', type: 'file' },
+    { path: 'd.js', type: 'file' },
+    { path: 'e.txt', type: 'file' },
+  ]
+  return nock('https://api.github.com')
+    .get('/repos/badges/shields/contents/services')
+    .reply(200, contents)
+}
+
+t.create('directory file count')
+  .get('/badges/shields/services.json')
+  .expectBadge({
+    label: 'files',
+    message: /^[0-9]+$/,
+  })
+
+t.create('directory file count (root)')
+  .get('/badges/shields.json')
+  .expectBadge({
+    label: 'files',
+    message: /^[0-9]+$/,
+  })
+
+t.create('directory file count (directory not found)')
+  .get('/badges/shields/not_existing_directory.json')
+  .expectBadge({
+    label: 'files',
+    message: 'repo or directory not found',
+  })
+
+t.create('directory file count (not a directory)')
+  .get('/badges/shields/package.json.json')
+  .expectBadge({
+    label: 'files',
+    message: 'not a directory',
+  })
+
+t.create('directory file count (dir type)')
+  .get('/badges/shields/services.json?type=dir')
+  .intercept(mockContents)
+  .expectBadge({
+    label: 'files',
+    message: '2',
+  })
+
+t.create('directory file count (file type)')
+  .get('/badges/shields/services.json?type=file')
+  .intercept(mockContents)
+  .expectBadge({
+    label: 'files',
+    message: '3',
+  })
```

Comment:
I think these (and the other mocked one at the bottom) would probably be better suited to unit tests vs. service tests with intercepts on mocks. For example from the Codecov service:

https://github.com/badges/shields/blob/master/services/codecov/codecov.spec.js

Reply:
Unit tests will cover only `transform` method logic but won't validate the correct values are passed to it by `handle` (namely `type` and `extension`). Therefore service tests provide greater confidence the code behaves as expected. If I wrote the unit tests for `transform` solely and then removed passing `{ type, extension }` to `transform` altogether, no test would turn red.

Reply:
I understand how the code flow works :wink: 

As stated above, our service tests are primarily used for validation of integrations with external endpoints and high level badge function. We integrate with a ton of platforms that can and do change their API contracts at will (we don't get notifications preemptively :smile:), and the service tests help us know whether or not we're still correctly interfacing with a service and able to render a badge.

We don't want to use service tests for automated verification of the interaction/call sites from one internal function to another. Again, I understand your reasoning and instinct, but it's a matter of tradeoffs. Within the Shields codebase and the contexts and constraints we work with, we favor and trust ourselves as maintainers to not merge and deploy such an obviously goofy change vs. utilizing our service tests to also verify all code-level component/function interactions.

This is fine for now, but please keep this in mind on any potential future PRs!

Reply:
I was not my intention to imply your lack of understanding. I was merely sharing my concerns :) Again, I have better understanding where you're coming from.

I've moved mocked service tests to unit tests.